src/spec-common/injectHeadless.ts

```diff
@@ -682,54 +687,50 @@ async function probeUserEnv(params: { defaultUserEnvProbe: UserEnvProbe; allowSy
 	}
 
 	if (env) {
-		await updateUserEnvCache(env, params.containerSessionDataFolder, containerProperties.shellServer);
+		await updateUserEnvCache(env, userEnvProbe, params, containerProperties.shellServer);
 	}
 
 	return env || {};
 }
 
-async function readUserEnvFromCache(cacheDir?: string, shellServer?: ShellServer) {
-	if (!shellServer || !cacheDir) {
+async function readUserEnvFromCache(userEnvProbe: UserEnvProbe, params: { output: Log; containerSessionDataFolder?: string }, shellServer?: ShellServer) {
+	if (!shellServer || !params.containerSessionDataFolder) {
 		return undefined;
 	}
 
-	const cacheFile = path.posix.join(cacheDir, 'env.json');
+	const cacheFile = path.posix.join(params.containerSessionDataFolder, `${userEnvProbe}.json`);
```

Comment:
`env-${userEnvProbe}.json` to make it clearer what it is about?

Reply:
makes sense. done